py-polars/polars/io/pyarrow_dataset/functions.py

```diff
@@ -37,6 +37,13 @@ def scan_pyarrow_dataset(
     This API is experimental and may change without it being considered a breaking
     change.
 
+    Notes
+    -----
+
+     - When using partitioning the appropriate `partitioning` option must be set on
+       `pyarrow.dataset.dataset` before passing to polars or the partitioned-on column(s)
+       may not get passed to polars.
```

Comment:
About Sphinx RST / numpydoc standard:

* Notes section should come before warnings
* No empty line after the start of a section
* If you have a single note, don't use the list syntax and it will render it with a nice "Note" syntax
* RST requires double backticks to get code style formatting
 
```suggestion
    Notes
    -----
    When using partitioning the appropriate ``partitioning`` option must be set on
    ``pyarrow.dataset.dataset`` before passing to polars or the partitioned-on column(s)
    may not get passed to polars.
```

(yes, we all dislike RST :))

Reply:
Sorry about the RST unfamiliarity, thanks. Regarding capitalization, I followed what was done in the rest of the file.

Reply:
If someone comes up with a nice linter for Sphinx RST, I would be so happy...